### ISSUE TITLE
lower-prio node flapping, master does not execute notify_master

### ISSUE THREAD

**Issue by gdanov** [acassen/keepalived#2087]
**Describe the issue**
I have 2 nodes with different priorities. For unclear reason the backup (lower prio) starts "flapping" and self-promotes to master for ~5 secs before self-demoting to backup again. The two nodes are in contact during this time. The master does not react in any way to this, does not change role or execute notify script.

My **question** is: is this expected behavior? Should I design my scripts with the expectation that one node may be promoted & demoted without the other node running it's own `notify_master`?

**To Reproduce**
unfortunately don't know how to trigger the backup node flapping again.

**Expected behavior**
Not expected, but **assumed** that when the two nodes are up, demotion of one would mandatorily trigger `notify_master` on the other.

**Keepalived version**
Keepalived v2.0.19 (10/19,2019)

**Distro (please complete the following information):**
	Ubuntu 20.04.3 LTS

**Details of any containerisation or hosted service (e.g. AWS)**
Hetzner VPS

** screenshot of the logs of the two nodes. Keep in mind this is pretty late into the ordeal and the backup node sanity is not guaranteed.

<img width="1270" alt="Screenshot 2022-01-28 at 15 03 52" src="https://user-images.githubusercontent.com/957113/151565196-a1a5c9ad-5bae-4211-a1e3-b833961fa747.png">

**Comment by pqarmitage** [comment]
What you are observing is usually caused by one of two reasons:

i) there is a break in network connectivity between the 2 nodes and the backup does not receive an advert from the master for over 3 advert intervals
ii) the backup node is very busy and keepalived does not get scheduled within the 3 and a bit advert intervals, and so when keepalived does run, it times the master out and takes over as master. This has been known to occur during backups, or other times of very high CPU load.
iii) keepalived is dumping its state (i.e. it has been sent SIGUSR1 or SIGUSR2) and /tmp is a slow device (e.g. and NFS mount).

It is correct that the master remains as the VRRP master since it hasn't received a higher priority advert.

It the cause of your problem is ii) there are various configuration options that can be specified to increase the priority of keepalived: `max_auto_priority`, `min_auto_priority_delay`, `vrrp_priority`, `vrrp_no_swap`, `vrrp_rt_priority`. See keepalived.conf(5) man page for more details.

There have been quite a number of reports of this type of issue in the past, and often people state that i) is not the case, only for them to discover after further investigation that they do have a network problem.

There is no known issue with keepalived that causes this.

**Comment by pqarmitage** [comment]
One sequence above to consider is:

At 13:09:03 the "backup" node times out and enters master state.
At 13:09:03, 13:09:04, 13:09:05, 13:09:06 and 13:09:07 the "master" node logs that it has received a lower priority advert (and it immediate sends its own (higher priority advert)).
It is not until 13:09:08 that the "backup" node logs that it has received a higher priority advert and reverts to backup state.

This suggests that the "backup" node is not receiving some of the adverts from the "master" node.

**Comment by gdanov** [comment]
#950 gives reasonable potential workaround. In the same ticket we see others too have lost packets when using VPC. I'll close this one.
